app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt

```diff
@@ -117,12 +118,16 @@ class ThreadFragment : Fragment() {
 
         val defaultTextColor = context.getColor(R.color.primaryTextColor)
         appBar.addOnOffsetChangedListener { appBarLayout, verticalOffset ->
-            val total = appBarLayout.height * COLLAPSE_TITLE_THRESHOLD
-            val removed = appBarLayout.height - total
-            val progress = (((-verticalOffset.toFloat()) - removed).coerceAtLeast(0.0) / total).toFloat() // Between 0 and 1
-            val opacity = (progress * 255).roundToInt()
 
+            val subjectHeight = appBarLayout.height.toDouble()
+            val impactingHeight = subjectHeight * COLLAPSE_TITLE_THRESHOLD
+            val nonImpactingHeight = subjectHeight - impactingHeight
+
+            val absoluteProgress = verticalOffset.absoluteValue.toDouble() - nonImpactingHeight
```

Comment:
This `toDouble()` is also not needed, same question

Reply:
https://github.com/Infomaniak/android-kMail/pull/881#discussion_r1172454057